### PR TITLE
Document GHC-76037

### PR DESCRIPTION
**Pull request by davidlaewen**

### CODE REVIEW

**Comment by davidlaewen** [comment]
Does this need further examples?

**Comment by david-christiansen** [comment]
The description sounds a lot like https://haskell.github.io/error-messages/messages/GHC-88464/index.html. Is the difference that this one applies to capitalized identifiers?

File: message-index/messages/GHC-76037/index.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+title: Not In Scope Error
+summary:  An identifier is not in scope
+introduced: 9.4.1
+severity: error
+---
+
+An identifier can only be used if it is in scope, i.e. if it is bound or declared previously and that binding/declaration applies to the usage location.
```
Comment (PPKFS) [inline message-index/messages/GHC-76037/index.md:8]:
```suggestion
An identifier can only be used if it is in scope; i.e. if it is bound or declared previously and that binding/declaration is available at the usage location.
```

I'm not sure, but I think "applies to the usage location" could be adjusted?

File: message-index/messages/GHC-76037/index.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+title: Not In Scope Error
+summary:  An identifier is not in scope
+introduced: 9.4.1
```
Comment (PPKFS) [inline message-index/messages/GHC-76037/index.md:4]:
```suggestion
introduced: 9.6.1
```

**Review by PPKFS — COMMENTED** [review]
Looking good -- maybe an example of where a constructor *should* be in scope but the user has forgotten it - for instance, importing `Foo` but not importing `Foo(..)` so `FooCtr` is not in scope?

File: message-index/messages/GHC-76037/index.md (hunk 1 above)
Comment (davidlaewen) [reply message-index/messages/GHC-76037/index.md:8]: Yes, I prefer this description

**Comment by davidlaewen** [comment]
> Looking good -- maybe an example of where a constructor _should_ be in scope but the user has forgotten it - for instance, importing `Foo` but not importing `Foo(..)` so `FooCtr` is not in scope?

Sounds good, will do :+1:

File: message-index/messages/GHC-76037/index.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+title: Not In Scope Error
+summary:  An identifier is not in scope
```
Comment (david-christiansen) [inline message-index/messages/GHC-76037/index.md:3]:
```suggestion
summary:  An identifier is not in scope.
```

File: message-index/messages/GHC-76037/index.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+title: Not In Scope Error
```
Comment (david-christiansen) [inline message-index/messages/GHC-76037/index.md:2]:
```suggestion
title: Not in scope
```

**Comment by davidlaewen** [comment]
> The description sounds a lot like https://haskell.github.io/error-messages/messages/GHC-88464/index.html. Is the difference that this one applies to capitalized identifiers?

I have the impression that there is some inconsistency with these two error codes, e.g.:
```
ghci> X

<interactive>:1:1: error: [GHC-88464]
    Data constructor not in scope: X
```
but
```
ghci> \X -> X

<interactive>:2:2: error: [GHC-76037]
    Not in scope: data constructor ‘X’
```

**Comment by david-christiansen** [comment]
Agh! Off I go to look at the GHC code...

**Comment by david-christiansen** [comment]
It seems that there's two ways to report out-of-scope idents: sometimes they are seen directly, and sometimes they're first thought to be holes and then later seen to not actually be holes. I'm not sure why it's that way, but it is. I think we should merge this, and then follow up later with GHC devs to see if these errors are in fact the same, semantically.

**Comment by davidlaewen** [comment]
Okay, that seems quite arbitrary from the end user perspective. But I guess we can always merge the two pages/errors into one later.

**Comment by david-christiansen** [comment]
I agree that this seems completely arbitrary. I think the thing to do is to merge this, and then follow up later with #116 .

**Review by PPKFS — APPROVED** [review]

**Pull request merged**